src/UserSpaceInstrumentation/include/UserSpaceInstrumentation/InstrumentProcess.h

```diff
@@ -32,12 +32,18 @@ class InstrumentationManager {
 
   [[nodiscard]] static std::unique_ptr<InstrumentationManager> Create();
 
+  struct InstrumentationResult {
+    absl::flat_hash_set<uint64_t> instrumented_function_ids;
+    absl::flat_hash_map<uint64_t, std::string> error_messages;
```

Comment:
Maybe `error_messages_by_function_id` or `function_ids_to_error_messages` or something like that for extra clarity?